Source/JavaScriptCore/dfg/DFGOperations.cpp

```diff
@@ -2806,6 +2810,10 @@ JSC_DEFINE_JIT_OPERATION(operationStringReplaceStringStringWithoutSubstitution,
     JITOperationPrologueCallFrameTracer tracer(vm, callFrame);
     auto scope = DECLARE_THROW_SCOPE(vm);
 
+    if (JSString* result = tryReplaceOneCharUsingString(globalObject, stringCell, searchCell, replacementCell))
```

Comment:
This code `operationStringReplaceStringStringWithoutSubstitution` is used when we ensured replacement string is not including substituion (`$`), so let's add template parameter to `tryReplaceOneCharUsingString` which avoids doing `$` search.